docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,594 @@
+
+
+Dataset helper / convenience methods
+------------------------------------
+We propose to implement a few convenience methods on all datasets. E.g.
+a `.residuals()` method, which computes the residual for the given data
+and state of the model:
+
+.. code::
+
+    dataset.residuals(method="")
+
+A selection of methods for the residual computation would ne available, such
+as `data - model`, `(data - model) / model` and `(data - model) / sqrt(model)`.
+
+For counts based datasets an `.excess()` method can be implmented.
+
+
+All dataset objects should implment a `__str__` method, so that the following
+works:
+
+.. code::
+
+    print(dataset)
+
+Specific to each dataset information on the model, model parameters, data and
+fit statistics is printed.
+
+We propose to implement a `.peek()` method on all datasets:
+
+.. code::
+
+    dataset.peek()
+
+In an interative environment this will show a default visualisation of the
+dataset including counts / data, IRFs, residuals etc.
+
+
+Simulation of `MapDataset` and `SpectrumDataset`
+------------------------------------------------
+The `MapDataset` and `SpectrumDataset` classes will be the central classes
+for map and spectrum based analyses. In many cases it's useful to simulate
+counts from a predicted number of counts model. The direct sampling of
+binned data for the datasets could be supported as following (illustrated
+for the `MapDataset`):
+
+.. code::
+
+    # allow counts=None on init
+    dataset = MapDataset(counts=None)
+
+    # to sample a counts map from npred
+    dataset.fake(random_seed=0)
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+Which calls `np.random.poisson()` internally on the npred map and sets it
+as a `.counts` map on the dataset. This works equivalently for the
+`SpectrumDataset`, `SpectrumDatasetOnOff` and `MapDatasetOnOff`. The on-off
+dataset require a background model for this to work.
+
+Unbinned simulation (sampling of event lists) is addressed in a separate PIG.
+
+
+`Datasets`
+----------
+
+To combine multiple datasets in a joint-likelihood analysis we propose to introduce
+a `Datasets` class. Its responsibility is to add up the log-likelihood values of
+the individual datasets and join the parameter lists. The `Datasets` class
+also represents the main interface to the `Fit` class.
+
+The following example shows how a joint-likelihood analysis of multiple observations
+is supposed to work:
+
+.. code ::
+
+    from gammapy.utils.fitting import Datatsets
+
+    model = SourceModel.read()
+
+    map_datasets = []
+
+    for obs_id in [123, 124, ...]:
+        bkg_model = BackgroundModel.read("obs-{}/background.fits".format(obs_id))
+        dataset = MapDataset(model=model, background=bkg_model)
+        map_datasets.append(dataset)
+
+    datasets = Datasets(map_datasets)
+
+    # check whether all contained datasets are of the same type
+    datasets.is_all_same_type
+
+    datasets.stat()
+
+    fit = Fit(datasets)
+
+    # or maybe simpler
```

Comment:
I think it's fine to allow passing Dataset or Datasets or List[Dataset] to `Fit`, and to always go to the uniform `Datasets`, which doesn't make dataset object copies, only possibly a new list, i.e. is cheap. Maybe mention this explicitly, since you show all three ways to interface to Fit in this PIG?